src/osd/PGLog.cc

```diff
@@ -86,24 +86,22 @@ void PGLog::IndexedLog::trim(
       }
     }
 
-    bool reset_complete_to = false;
-    // we are trimming past complete_to, so reset complete_to
-    if (complete_to != log.end() && e.version >= complete_to->version)
-      reset_complete_to = true;
     if (rollback_info_trimmed_to_riter == log.rend() ||
 	e.version == rollback_info_trimmed_to_riter->version) {
       log.pop_front();
       rollback_info_trimmed_to_riter = log.rend();
     } else {
       log.pop_front();
     }
-
-    // reset complete_to to the beginning of the log
-    if (reset_complete_to) {
-      lgeneric_subdout(cct, osd, 20) << " moving complete_to " << " to "
-                      << log.begin()->version << dendl;
```

Comment:
@neha-ojha I mean it is not safe to dereference log.begin() here since log could be empty. Also my way of resetting  complete_to is more efficient if there are tons of log entries to trim, right?